Autodesk/synthesis
# Engine Analytics

Google analytics have been incorporated in the engine in the following areas:
- Button click events on main buttons
- Additional button tracking on sensor menus
- Time tracking on a user's session until they reload the scene (via field reset, exiting, etc.)
- Time tracking on how long a user spends on a specific toolbar tab

This addresses JIRA Issues [#988](https://jira.autodesk.com/browse/AARD-988) and [#1041](https://jira.autodesk.com/browse/AARD-1041).

**Main Toolbar:** Each button on this toolbar has a click event
![image](https://user-images.githubusercontent.com/16855850/61072876-135b1100-a3c9-11e9-804d-59351ac7d653.png)

**DPM Toolbar:** Each button on this toolbar has a click event
![image](https://user-images.githubusercontent.com/16855850/61072922-2837a480-a3c9-11e9-86b5-9857e8a9fbdd.png)

**Scoring Toolbar:** Each button on this toolbar has a click event
![image](https://user-images.githubusercontent.com/16855850/61072950-34bbfd00-a3c9-11e9-8fdd-e045ccf5955a.png)

**Sensor Toolbar:** Additional sensor events tracked to identify which sensors users are interested in.
![image](https://user-images.githubusercontent.com/16855850/61072971-42718280-a3c9-11e9-9a75-af46b7e3565b.png)
![image](https://user-images.githubusercontent.com/16855850/61072999-52896200-a3c9-11e9-8e83-c4f43cd42210.png)

**Emulation Toolbar:** Each button on this toolbar has a click event
![image](https://user-images.githubusercontent.com/16855850/61073042-6765f580-a3c9-11e9-8116-295f81264167.png)

**Time Tracking on Tabs**
![image](https://user-images.githubusercontent.com/16855850/61073075-751b7b00-a3c9-11e9-9b42-fd7a7ffd0157.png)

Summary
:white_check_mark: Real-time tracking works for the main buttons as described above. 
:large_orange_diamond: Logging events is delayed at least 4-6 hours so we need people clicking on the toolbar buttons and tabs so I can look through the uploaded data and verify timing is functioning. 

To Test
- Verify building Unity works and no compile errors exist.
- Gain access to the team Google Analytics account (please contact Kira, Andrei, or Katie for the account login.)
- Once logged in, verify you are accessing the Unity project: 
![image](https://user-images.githubusercontent.com/16855850/61073383-25897f00-a3ca-11e9-9162-df8ea00ed73a.png)
- On the left-hand toolbar, click on "Realtime" then "Events." 
![image](https://user-images.githubusercontent.com/16855850/61073426-4356e400-a3ca-11e9-9035-6bebc1117dae.png)

Note: The realtime tracking data has some delay but click on a few buttons and allow some time for update. ONLY main buttons have tracking data. For an example, anything on a toolbar is tracked but an "X" button or buttons within panels have not been tracked.
